src/NuGet.Core/NuGet.Commands/RestoreCommand/Utility/BuildAssetsUtils.cs

```diff
@@ -22,428 +22,698 @@ namespace NuGet.Commands
+            return files;
+        }
 
-            // Get assets for each framework
-            foreach (var projectFramework in project.TargetFrameworks.Select(f => f.FrameworkName))
+        public static string ReplacePathsWithMacros(string path)
+        {
+            foreach (var macroName in MacroCandidates)
             {
-                var targetsAndProps =
-                    GetTargetsAndPropsForFramework(
-                        targetGraphs,
-                        repositories,
-                        context,
-                        request,
-                        includeFlagGraphs,
-                        projectFramework);
+                string macroValue = Environment.GetEnvironmentVariable(macroName);
+                if (!string.IsNullOrEmpty(macroValue)
+                    && path.StartsWith(macroValue, StringComparison.OrdinalIgnoreCase))
+                {
+                    path = $"$({macroName})" + $"{path.Substring(macroValue.Length)}";
+                }
+
+                break;
+            }
+
+            return path;
+        }
+
+        public static XDocument GenerateMultiTargetFrameworkWarning()
+        {
+            var doc = GenerateEmptyImportsFile();
+
+            doc.Root.Add(new XElement(Namespace + "Target",
+                        new XAttribute("Name", "EmitMSBuildWarning"),
+                        new XAttribute("BeforeTargets", "Build"),
+
+                        new XElement(Namespace + "Warning",
+                            new XAttribute("Text", Strings.MSBuildWarning_MultiTarget))));
+
+            return doc;
+        }
+
+        /// <summary>
+        /// Add standard properties to only props file if it exists, otherwise the targets.
+        /// </summary>
+        public static void AddNuGetPropertiesToFirstImport(IEnumerable<MSBuildOutputFile> files,
+            IEnumerable<string> packageFolders,
+            string repositoryRoot,
+            RestoreOutputType outputType,
+            bool success)
+        {
+            var firstImport = files.Where(file => file.Content != null)
+                .OrderByDescending(file => file.Path.EndsWith(PropsExtension) ? 1 : 0)
+                .FirstOrDefault();
 
-                buildAssetsByFramework.Add(projectFramework, targetsAndProps);
+            if (firstImport != null)
+            {
+                AddNuGetProperties(firstImport.Content, packageFolders, repositoryRoot, outputType, success);
             }
+        }
 
-            var props = new List<MSBuildRestoreImportGroup>();
-            var targets = new List<MSBuildRestoreImportGroup>();
+        /// <summary>
+        /// Apply standard properties in a property group.
+        /// </summary>
+        public static void AddNuGetProperties(XDocument doc, IEnumerable<string> packageFolders, string repositoryRoot, RestoreOutputType outputType, bool success)
+        {
+            var projectStyle = "Unknown";
```

Comment:
I like the enum idea. This will be changed in a different PR to cover https://github.com/NuGet/Home/issues/4047 and several similar issues where these values will now be used across the SDK, NuGet, and packages. This property had a different goal originally.